tests/internal/import_hooks/test_decorator.py

```diff
@@ -0,0 +1,56 @@
+import mock
+import pytest
+
+from ddtrace.internal import import_hooks
+from ddtrace.internal.import_hooks.decorator import register_module_hook
+
+
+@pytest.fixture
+def hooks():
+    try:
+        yield import_hooks.hooks
```

Comment:
You don't have to test and use the global hooks object you created for all tests. That could become a problem once e.g. you want to run your tests in parallel.
Don't use a fixture here, just create a `ModuleHookRegistry` in the test that use one.

Reply:
Yeah, the problem I have is that the patching code and the `register_module_hook` decorator uses just the global `ModuleHookRegistry`, so probably indicates an area where the overall design needs improvement then.